consensus/weight_test.go

```diff
@@ -0,0 +1,156 @@
+package consensus_test
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ipfs/go-hamt-ipld"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/state"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestNewWeight(t *testing.T) {
+	cst := hamt.NewCborStore()
+	ctx := context.Background()
+	fakeTree := &state.MockStateTree{}
+	// We only care about total power for the weight function
+	// Total is 16, so bitlen is 5
+	as := consensus.NewFakeActorStateStore(types.NewBytesAmount(1), types.NewBytesAmount(16), make(map[address.Address]address.Address))
+	tickets := []types.Ticket{consensus.MakeFakeTicketForTest()}
+	toWeigh := types.RequireNewTipSet(t, &types.Block{
+		ParentWeight: 0,
+		Tickets:      tickets,
+	})
+	sel := consensus.NewChainSelector(cst, as, types.CidFromString(t, "genesisCid"))
+
+	t.Run("basic happy path", func(t *testing.T) {
+		// 0 + 1[2*1 + 5] = 7
+		fixWeight, err := sel.NewWeight(ctx, toWeigh, fakeTree)
+		assert.NoError(t, err)
+		assertEqualInt(t, 7, fixWeight)
```

Comment:
Later it's going to be difficult to decode this 7 and the equation in the comment above. It'd be better to put the equation as the expected, populate it will all the known constants, create variables for all the expected values.